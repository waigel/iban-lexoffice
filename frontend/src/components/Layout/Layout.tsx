import React from 'react';
import {Footer} from '../Footer/Footer';

export interface LayoutProps {
  children: React.ReactElement;
}

export const Layout = ({children}: LayoutProps) => {
  return (
    <div className="w-screen h-screen bg-black text-white">
      <main>{children}</main>
      <Footer />
    </div>
  );
};
