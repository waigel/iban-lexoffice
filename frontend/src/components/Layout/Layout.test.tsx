import React from 'react';
import TestRenderer from 'react-test-renderer';
import {Layout} from './Layout';

jest.mock('../Footer/Footer', () => ({
  Footer: () => <div>Footer</div>,
}));

it('should test the Layout component is rendered correctly', () => {
  const tree = TestRenderer.create(
    <Layout>
      <p>TEST CHILD</p>
    </Layout>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
