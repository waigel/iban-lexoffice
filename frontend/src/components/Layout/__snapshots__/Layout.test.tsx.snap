// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Layout component is rendered correctly 1`] = `
<div
  className="w-screen h-screen bg-black text-white"
>
  <main>
    <p>
      TEST CHILD
    </p>
  </main>
  <div>
    Footer
  </div>
</div>
`;
